app/shared/github_handler.rb

```diff
@@ -42,6 +44,24 @@ def self.included(klass)
 
     def github_action(client, &block)
       if client.kind_of?(Octokit::Client)
+        # Inject Faraday::Request::Retry to the client if necessary
+        unless client.middleware.handlers.include?(Faraday::Request::Retry)
```

Comment:
I think it could be worthwhile to pull this out into a separate method instead of inline in the `github_action` method.